### PR TITLE
[Doc] Add descriptions for namespace isolation policy, configuration store, and metadata store

### PR DESCRIPTION
**Pull request by Anonymitaet**

### CODE REVIEW

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaceIsolationPolicy.java [hunk 1]
```diff
@@ -46,7 +46,7 @@
         @Parameter(names = "--namespaces", description = "comma separated namespaces-regex list", required = true, splitter = CommaParameterSplitter.class)
         private List<String> namespaces;
 
-        @Parameter(names = "--primary", description = "comma separated  primary-broker-regex list", required = true, splitter = CommaParameterSplitter.class)
+        @Parameter(names = "--primary", description = "comma separated  primary-broker-regex list. Before topics are assigned to brokers, you can set the namespace isolation policy with a primary or secondary regex to select desired brokers. If no broker matches the specified regex, you cannot create a topic. If the number of primary brokers are not enough, topics are assigned to secondary brokers. If the number of secondary brokers are not enough, topics are assigned to other brokers which do not have any isolation policies.", required = true, splitter = CommaParameterSplitter.class)
```
Comment (Huanli-Meng) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdNamespaceIsolationPolicy.java:49]:
```suggestion
        @Parameter(names = "--primary", description = "comma separated  primary-broker-regex list. Before topics are assigned to brokers, you can set the namespace isolation policy with a primary or secondary regex to select desired brokers. If no broker matches the specified regex, you cannot create a topic. If there are not enough primary brokers, topics are assigned to secondary brokers. If there are not enough secondary brokers, topics are assigned to other brokers which do not have any isolation policies.", required = true, splitter = CommaParameterSplitter.class)
```

File: site2/docs/concepts-architecture-overview.md [hunk 2]
```diff
@@ -45,11 +45,17 @@ Clusters can replicate amongst themselves using [geo-replication](concepts-repli
 
 ## Metadata store
 
-Pulsar uses [Apache Zookeeper](https://zookeeper.apache.org/) for metadata storage, cluster configuration, and coordination. In a Pulsar instance:
+The Pulsar metadata store maintains all the metadata of a Pulsar cluster, such as topic metadata, schema, broker load data, and so on. Pulsar uses [Apache Zookeeper](https://zookeeper.apache.org/) for metadata storage, cluster configuration, and coordination. The Pulsar metadata store can be deployed on a separate ZooKeeper cluster or deployed on an existing ZooKeeper cluster.
```
Comment (Huanli-Meng) [inline site2/docs/concepts-architecture-overview.md:48]: Zookeeper or ZooKeeper, should be consistent.

**Review by Huanli-Meng — COMMENTED** [review]
leave my comments, PTAL

**Comment by Anonymitaet** [comment]
@Huanli-Meng thanks for your comments, I've incorporated, PTAL.

**Review by Jennifer88huang — APPROVED** [review]

**Review by sijie — APPROVED** [review]

**Pull request merged**